HW04.Rmd

```diff
@@ -1,126 +1,273 @@
-Finally, all together now:
-```{r police_3}
-
-```
-
-
-## Session info
-
-```{r}
-# always good to have this for reproducibility purposes
-devtools::session_info()
-```
-
+---
+title: "Exploring Gun Deaths in America"
+output: github_document
+---
+
+## Objectives
+* To explore gun death data curated by FiveThirtyEight [original data available here](https://github.com/fivethirtyeight/guns-data) by
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(dplyr)
+library(magrittr)
+library(knitr)
+
+# read in the data
+gun_death<- read_csv("gun_deaths.csv")
+
+```
+```{r setup, include=TRUE}
+knitr::opts_chunk$set(echo = TRUE, error = TRUE)
+
+#glimps
+glimpse(gun_death)
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+```{r month, warning=FALSE}
+month_death<- gun_death %>%
+    group_by(month) %>%
+    count(month)
+
+kable(month_death, format="markdown", row.names= NA, col.names= c("Month", "Number of Deaths"), caption= "Monthly gun-related deaths in 2012, data retrieved from https://github.com/fivethirtyeight/guns-data")
+```
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+#change variable month to contain abbs of months
+gun_death$month <- factor(gun_death$month, labels= c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"))
+
+#create a data frame that has the count of death for each month 
+monthly_death<- gun_death %>%
+    group_by(month) %>%
+    count()
+
+#plot   
+ggplot(monthly_death, aes(x=month, y=n))+
+    geom_bar(stat="identity", fill="red", alpha=0.7)+
+    scale_y_continuous(limit=c(0,12000))+
+    labs(title="Montly Gun Deaths in 2012", x=element_blank(), y="Number of deaths")+
+    theme_bw()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#create a data frame with death number associated with intent type
+gun_death_intent<- gun_death %>%
+    group_by(intent) %>%
+    count() 
+
+#A way to level the intent variable in order to order the bins, not the best way if there are more variables
+gun_death_intent$intent<- factor(gun_death_intent$intent, 
+                                 levels=c("Suicide", "Homicide", "Accidental", "Undetermined", "Not Available"))     
```

Comment:
I think another way to order the bins is using function fct_infreq( ).

Reply:
That's a useful function to know, thanks!